File: 119. Pascal's Triangle II/119. Pascal's Triangle II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> row(rowIndex + 1, 1);
        
        for (int i = 1; i <= rowIndex; i++) {
            long long prev = row[0];
            for (int j = 1; j < i; j++) {
                long long temp = row[j];
                row[j] = (prev * (rowIndex - j + 1)) / j;
                prev = temp;
            }
            row[i] = 1;
        }
        
        return row;
    }
};

File: 119. Pascal's Triangle II/119. Pascal's Triangle II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> row = new ArrayList<>(rowIndex + 1);
        for (int i = 0; i <= rowIndex; i++) {
            row.add(1);
        }
        
        for (int i = 1; i <= rowIndex; i++) {
            long prev = row.get(0);
            for (int j = 1; j < i; j++) {
                long temp = row.get(j);
                row.set(j, (int)((prev * (rowIndex - j + 1)) / j));
                prev = temp;
            }
        }
        
        return row;
    }
}